src/panels/lovelace/editor/config-elements/hui-heading-card-editor.ts

```diff
@@ -0,0 +1,354 @@
+              selector: {
+                ui_action: {
+                  default_action: "none",
+                  actions,
+                },
+              },
+            },
+          ],
+        },
+      ] as const satisfies readonly HaFormSchema[]
+  );
+
+  private _entitySchema = memoizeOne(
+    () =>
+      [
+        {
+          name: "entity",
+          selector: { entity: {} },
+        },
+        {
+          name: "icon",
+          selector: { icon: {} },
+          context: { icon_entity: "entity" },
+        },
+        {
+          name: "content",
+          selector: { ui_state_content: {} },
+          context: { filter_entity: "entity" },
+        },
+        {
+          name: "interactions",
+          type: "expandable",
+          flatten: true,
+          iconPath: mdiGestureTap,
+          schema: [
+            {
+              name: "tap_action",
+              selector: {
+                ui_action: {
+                  default_action: "none",
+                },
+              },
+            },
+          ],
+        },
+      ] as const satisfies readonly HaFormSchema[]
+  );
+
+  protected render() {
+    if (!this.hass || !this._config) {
+      return nothing;
+    }
+
+    return cache(
+      this._entityFormEditorData ? this._renderEntityForm() : this._renderForm()
+    );
+  }
+
+  private _renderEntityForm() {
+    const schema = this._entitySchema();
+    return html`
+      <hui-sub-form-editor
+        .label=${this.hass!.localize(
+          "ui.panel.lovelace.editor.entities.form-label"
+        )}
+        .hass=${this.hass}
+        .data=${this._entityFormEditorData!.data}
+        @go-back=${this._goBack}
+        @value-changed=${this._subFormChanged}
+        .schema=${schema}
+        .assertConfig=${this._assertEntityConfig}
+        .computeLabel=${this._computeEntityLabelCallback}
+      >
+      </hui-sub-form-editor>
+    `;
+  }
+
+  private _entities = memoizeOne((entities: HeadingCardConfig["entities"]) =>
+    processEditorEntities(entities || [])
+  );
+
+  private _renderForm() {
+    const data = {
+      ...this._config!,
+    };
+
+    if (!data.heading_style) {
+      data.heading_style = "title";
+    }
+
+    const schema = this._schema(this.hass!.localize);
+
+    return html`
+      <ha-form
+        .hass=${this.hass}
+        .data=${data}
+        .schema=${schema}
+        .computeLabel=${this._computeLabelCallback}
+        @value-changed=${this._valueChanged}
+      ></ha-form>
+      <ha-expansion-panel outlined>
+        <h3 slot="header">
+          <ha-svg-icon .path=${mdiListBox}></ha-svg-icon>
+          ${this.hass!.localize(
+            "ui.panel.lovelace.editor.card.heading.entities"
+          )}
+        </h3>
+        <div class="content">
+          <hui-entities-editor
+            .hass=${this.hass}
+            .entities=${processEditorEntities(
+              this._entities(this._config!.entities)
+            )}
```

Comment:
it is processed twice now
```suggestion
            .entities=${this._entities(this._config!.entities)}
```